core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1698,7 +1708,7 @@ class KafkaController(val config: KafkaConfig,
   private def processPartitionModifications(topic: String): Unit = {
     def restorePartitionReplicaAssignment(
       topic: String,
-      newPartitionReplicaAssignment: Map[TopicPartition, ReplicaAssignment]
+      newPartitionReplicaAssignment: Set[(TopicPartition, ReplicaAssignment)]
```

Comment:
Why did we make this a set?

Reply:
we changed the call to Zk from `zkClient.getFullReplicaAssignmentForTopics` to `zkClient.getReplicaAssignmentAndTopicIdForTopics` since the former dud not provide the topic Id. As a result of this change we needed to use `flatMap` at line 1730 and hence it results in a `Set`.

We can work on improving the readibility as you suggested once we have a consensus on whether we even want to make this change at all.